File: src/main/java/de/thatsich/intellie/common/module/item/IItemArmor.java
package de.thatsich.intellie.common.module.item;

import net.minecraft.item.Item;
import net.minecraft.item.ItemArmor;
import net.minecraft.item.ItemStack;

/**
 @author thatsIch
 @date 21.03.2014. */
public interface IItemArmor extends IItem
{
	/**
	 Return the armor material for this armor item.
	 */
	ItemArmor.ArmorMaterial getArmorMaterial ();

	/**
	 Return whether the specified armor ItemStack has a color.
	 */
	boolean hasColor ( ItemStack par1ItemStack );

	/**
	 Return the color for the specified armor ItemStack.
	 */
	int getColor ( ItemStack par1ItemStack );

	/**
	 Remove the color from the specified armor ItemStack.
	 */
	void removeColor ( ItemStack par1ItemStack );

	void func_82813_b ( ItemStack par1ItemStack, int par2 );

	@Override
	Item setContainerItem ( Item par1Item );
}

File: src/main/scala/de/thatsich/minecraft/core/module/item/IItemArmor.scala
package de.thatsich.minecraft.core.module.item

import net.minecraft.item.{ItemArmor, ItemStack, Item}

/**
 *
 *
 * @author thatsIch
 * @since 07.04.2014.
 */
trait IItemArmor extends IItem
{
	/**
	Return the armor material for this armor item.
	  */
	def getArmorMaterial: ItemArmor.ArmorMaterial

	/**
	Return whether the specified armor ItemStack has a color.
	  */
	def hasColor(par1ItemStack: ItemStack): Boolean

	/**
	Return the color for the specified armor ItemStack.
	  */
	def getColor(par1ItemStack: ItemStack): Int

	/**
	Remove the color from the specified armor ItemStack.
	  */
	def removeColor(par1ItemStack: ItemStack)

	def func_82813_b(par1ItemStack: ItemStack, par2: Int)

	@Override def setContainerItem(par1Item: Item): Item
}
